lib/MCT/Controller/Conference.pm

```diff
@@ -2,6 +2,42 @@ package MCT::Controller::Conference;
 
 use Mojo::Base 'Mojolicious::Controller';
 
+sub buy {
+  my $c = shift;
+  my $product = $c->model->conference_product(id => $c->param('product_id') || '');
+  my $user = $c->model->user(id => $c->session('uid'));
+  my $validation = $c->validation;
+
+  return $c->register('Missing stripeToken') unless $c->param('stripeToken');
+  return $c->register('Missing stripeEmail') unless $c->param('stripeEmail');
+  $user->id($c->session('cid')) if $c->session('cid');
+
+  $c->delay(
+    sub { $product->load(shift->begin) },
+    sub {
+      my ($delay, $err) = @_;
+      die $err if $err;
+      return $c->register("Unknown product.") unless $product->name;
+      return $c->register("Product '@{[$product->name]}' has been changed.") unless $product->same_as($c->req->body_params->to_hash);
+      return $user->load($delay->begin);
+    },
+    sub {
+      my ($delay, $err) = @_;
+      die $err if $err;
+      return $c->register('Email mismatch.') unless $user->email eq $c->param('stripeEmail');
+      return $c->stripe->charge({description => $product->description}, $delay->begin);
+    },
+    sub {
```

Comment:
I'm curious if there's a big chance we can run stripe->charge, but fail to do add_product(). I'm not even sure how to write this with a transaction to fix it.